File: android/RootStream/app/src/main/kotlin/com/rootstream/data/models/Peer.kt
package com.rootstream.data.models

import java.util.UUID

/**
 * Peer model representing a discovered streaming host
 */
data class Peer(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val hostname: String,
    val port: Int,
    val ipAddress: String,
    val capabilities: List<String> = emptyList(),
    val lastSeen: Long = System.currentTimeMillis(),
    val isOnline: Boolean = true
) {
    val displayName: String
        get() = "$name ($ipAddress:$port)"
}

File: ios/RootStream/RootStream/Models/Peer.swift
//
//  Peer.swift
//  RootStream iOS
//
//  Model representing a discovered peer
//

import Foundation

struct Peer: Identifiable, Codable {
    let id: String
    let name: String
    let hostname: String
    let port: UInt16
    let isManual: Bool
    var lastSeen: Date?
    var publicKey: String?
    var deviceInfo: DeviceInfo?
    
    init(id: String, name: String, hostname: String, port: UInt16, isManual: Bool = false) {
        self.id = id
        self.name = name
        self.hostname = hostname
        self.port = port
        self.isManual = isManual
        self.lastSeen = Date()
    }
}

struct DeviceInfo: Codable {
    let osVersion: String
    let deviceModel: String
    let capabilities: [String]
}
